test/e2e/monitoring_stack_controller_test.go

```diff
@@ -163,6 +167,24 @@ func reconcileRevertsManualChanges(t *testing.T) {
 	assert.DeepEqual(t, generated.Spec, reconciled.Spec)
 }
 
+func recreateDeleteGrafanaDS(t *testing.T) {
+	ms := newMonitoringStack(t, "test-grafana-ds")
+	datasourceName := fmt.Sprintf("ms-%s-%s", ms.Namespace, ms.Name)
+
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+
+	grafanaDS := grafanav1alpha1.GrafanaDataSource{}
+
+	f.GetResourceWithRetry(t, datasourceName, "monitoring-stack-operator", &grafanaDS)
```

Comment:
"monitoring-stack-operator" -> grafanaop.Namespace

Reply:
Is it possible to  make use of `AssertResourceEventuallyExists` ? to avoid Get & assert?

Reply:
`AssertResourceEventuallyExists` is created for statefulset,  there will need to be some refactoring done to make it work.